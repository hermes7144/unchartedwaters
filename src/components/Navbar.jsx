import { React } from 'react';
import { Link } from 'react-router-dom';
import { BsFillPencilFill } from 'react-icons/bs';
import User from './User';
import Button from './ui/Button';
import { useAuthContext } from './context/AuthContext';

export default function Navbar() {
  const { user, login, logout } = useAuthContext();
  return (
    <header className='flex justify-between border-b border-gray-300 p-2'>
      <Link to='/' className='flex items-center text-4xl text-brand'>
        <h1>DH Calc.</h1>
      </Link>
      <nav className='flex items-center gap-4 font-semibold'>
        <Link to='/invest'>Invest</Link>

        <Link to='/general'>General</Link>
        <Link to='/info'>Info</Link>
        <Link to='/qna'>QnA</Link>
        {user && user.isAdmin && (
          <Link to='/writing/new'>
            <BsFillPencilFill />
          </Link>
        )}

        {user && <User user={user} />}
        {!user && <Button text={'Login'} onClick={login}></Button>}
        {user && <Button text={'Logout'} onClick={logout}></Button>}
      </nav>
    </header>
  );
}
