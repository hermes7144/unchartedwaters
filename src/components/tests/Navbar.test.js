import { Route } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { withRouter } from '../../tests/utils'
import Navbar from '../Navbar'

describe('SearchHeader', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      withRouter(<Route path='/' element={<Navbar />} />)
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
})