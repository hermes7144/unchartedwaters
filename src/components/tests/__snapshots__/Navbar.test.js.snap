// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader renders correctly 1`] = `
<header
  className="flex justify-between border-b border-gray-300 p-2"
>
  <a
    className="flex items-center text-4xl text-brand"
    href="/"
    onClick={[Function]}
  >
    <h1>
      DH Calc.
    </h1>
  </a>
  <nav
    className="flex items-center gap-4 font-semibold"
  >
    <a
      href="/invest"
      onClick={[Function]}
    >
      Invest
    </a>
    <a
      href="/general"
      onClick={[Function]}
    >
      General
    </a>
    <a
      href="/citys"
      onClick={[Function]}
    >
      Citys
    </a>
    <a
      href="/qna"
      onClick={[Function]}
    >
      QnA
    </a>
  </nav>
</header>
`;
